tests/test_splitter.py

```diff
@@ -0,0 +1,388 @@
+            )
+            iterable.append(item)
+
+        dataset = Dataset.from_iterable(iterable, categories)
+        return dataset, totals
+
+    def test_split_for_detection(self):
+        def append_bbox_coco(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                id=kwargs['ann_id'],
+                                attributes=kwargs['attributes'],
+                                group=kwargs['label_id']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_voc(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                id=kwargs['ann_id']+1,
+                                attributes=kwargs['attributes'],
+                                group=kwargs['label_id'])) # obj
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id']+3,
+                                group=kwargs['label_id'])) # part
+            annotations.append(Label(kwargs['label_id']+3,
+                                attributes=kwargs['attributes']))
+        def append_bbox_yolo(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_cvat(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                id=kwargs['ann_id'],
+                                attributes=kwargs['attributes'],
+                                group=kwargs['label_id'],
+                                z_order=kwargs['ann_id']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_labelme(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                attributes=kwargs['attributes'],
+                                id=kwargs['ann_id']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_mot(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+            annotations.append(Label(kwargs['label_id'],
+                                attributes=kwargs['attributes']))
+        def append_bbox_widerface(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, attributes=kwargs['attributes']))
+            annotations.append(Label(0, attributes=kwargs['attributes']))
+
+        functions = {
```

Comment:
I think, this kind of testing is hard to maintain, but let's leave it as is. However, a generic test should also be added.

Reply:
Can you share your thought more on the generic test?

Reply:
Basically, the same, as these, but with the most complete inputs. For example - full instances from COCO (bbox or polygons, or mask + points, if it matters, in a group) with a couple of attributes and dataset items - depending on the algorithm. You can also look at test coverage, if you need some hints.

Reply:
I see. Do you think I should enter more complicated test sets even though the DetectionSplit (I changed the class name) only cares Bbox annotations?

Reply:
No, there is no need in that.